### PR TITLE
feat: Better error messages (command not found + cleaner logs)

### PR DESCRIPTION
**Pull request by juanig1**
Work a bit on a better user experience
- Add `oclif-plugin-not-found`
  - Suggest similar commands if original command input does not exist

### CODE REVIEW

File: bin/run [hunk 1]
```diff
@@ -36,9 +37,15 @@ async function main() {
   } catch(err) {
     await require('@oclif/errors/handle')(err);
 
-    console.log(err.stack);
-    console.log(err.code);
     console.log(err.message);
+
```
Comment (hassy) [inline bin/run:41]: I suggest this is reverted. The error message printed here is important information for the user. We should not require a debug flag to be set for them to see it.

File: test/testcases/errors-and-warnings.bats [hunk 2]
```diff
@@ -31,9 +31,9 @@
   [[ true ]]
 }
 
-@test "Exits with non zero when an unknown command is used" {
-    run ./bin/run makemeasandwich --with cheese
-    [[ $status -ne 0 ]]
+@test "Suggest similar commands if unknown command is used" {
+   output=echo "n" | ./bin/run helpp 2>&1 | grep -q "Did you mean help"
+   [[ $output -eq 0 ]]
```
Comment (hassy) [inline test/testcases/errors-and-warnings.bats:36]: What happens when a command that isn't similar to any existing command is used? We still need a test for that.

Comment (juanig1) [reply test/testcases/errors-and-warnings.bats:36]: it's [not configurable](https://github.com/oclif/plugin-not-found/blob/main/src/index.ts#L15) sadly. I kind of expected it to discard suggestions where dist > N, but no

Comment (hassy) [reply test/testcases/errors-and-warnings.bats:36]: Okay, cool 👍 Let's merge this!

**Comment by juanig1** [comment]
moved here https://github.com/artilleryio/artillery/pull/1699

**Pull request closed without merging**